docs/DeveloperGuide.md

```diff
@@ -154,6 +154,30 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Sort feature
+
+#### Implementation
+
+<img src="images/SortCommandDiagram.png" width="574" />
+
+The sort command is facilitated by `ModelManager`. `ModelManager` contains a `SortedList<Person>` and 
+`FilteredList<Person>` to provide sorting and filtering on the same list. The original list is fetched from 
+`AddressBook#getPersonList()` and passed to the `FilteredList`. The `FilteredList` is then passed to the `SortedList`.
+By returning the `SortedList` as an `ObservableList<Person>`, the UI will be able to update the person cards displayed
+after the `find` or `sort` command has been used.
+
+To facilitate comparing between two `Person` objects, the attributes also implement `Comparable<Person>`. The 
+`PersonCompmarators` class contains the comparators that can be used to compare persons based on the attribute.
+
+The command only allows for the sorting to be performed on one of the Person's attribute. The `SortCommandParser` will
+extract the attribute, and create a `SortCommand` using the corresponding `Comparator<Person>` for that attribute.
+
+#### Why it is implemented this way
```

Comment:
Could we use a header here so it is more clear?
e.g. 
#### Design considerations:
**Aspect: using static comparators compared to lambda functions**